worlds/wl4/__init__.py

```diff
@@ -0,0 +1,240 @@
+        'Palm Tree Paradise': set(get_level_locations(Passage.EMERALD, 0)),
+        'Wildflower Fields': set(get_level_locations(Passage.EMERALD, 1)),
+        'Mystic Lake': set(get_level_locations(Passage.EMERALD, 2)),
+        'Monsoon Jungle': set(get_level_locations(Passage.EMERALD, 3)),
+        'Cractus Treasures': set(k for k, v in get_level_location_data(Passage.EMERALD, 4) if v.source != LocationType.BOSS),
+        'The Curious Factory': set(get_level_locations(Passage.RUBY, 0)),
+        'The Toxic Landfill': set(get_level_locations(Passage.RUBY, 1)),
+        '40 Below Fridge': set(get_level_locations(Passage.RUBY, 2)),
+        'Pinball Zone': set(get_level_locations(Passage.RUBY, 3)),
+        'Cuckoo Condor Treasures': set(k for k, v in get_level_location_data(Passage.RUBY, 4) if v.source != LocationType.BOSS),
+        'Toy Block Tower': set(get_level_locations(Passage.TOPAZ, 0)),
+        'The Big Board': set(get_level_locations(Passage.TOPAZ, 1)),
+        'Doodle Woods': set(get_level_locations(Passage.TOPAZ, 2)),
+        'Domino Row': set(get_level_locations(Passage.TOPAZ, 3)),
+        'Aerodent Treasures': set(k for k, v in get_level_location_data(Passage.TOPAZ, 4) if v.source != LocationType.BOSS),
+        'Crescent Moon Village': set(get_level_locations(Passage.SAPPHIRE, 0)),
+        'Arabian Night': set(get_level_locations(Passage.SAPPHIRE, 1)),
+        'Fiery Cavern': set(get_level_locations(Passage.SAPPHIRE, 2)),
+        'Hotel Horror': set(get_level_locations(Passage.SAPPHIRE, 3)),
+        'Catbat Treasures': set(k for k, v in get_level_location_data(Passage.SAPPHIRE, 4) if v.source != LocationType.BOSS),
+        'Golden Passage': set(get_level_locations(Passage.GOLDEN, 0)),
+    }
+
+    web = WL4Web()
+
+    def generate_early(self):
+        if self.options.goal in (Goal.option_local_golden_treasure_hunt, Goal.option_local_golden_diva_treasure_hunt):
+            self.options.local_items.value.update(self.item_name_groups['Golden Treasure'])
+        if self.options.required_jewels > self.options.pool_jewels:
+            self.options.pool_jewels = PoolJewels(self.options.required_jewels)
+        if self.options.required_jewels >= 1 and self.options.golden_jewels == 0:
+            self.options.golden_jewels = GoldenJewels(1)
+
+    def create_regions(self):
+        location_table = self.setup_locations()
+        create_regions(self, location_table)
+        set_access_rules(self)
+        connect_regions(self)
+
+        passages = ('Emerald', 'Ruby', 'Topaz', 'Sapphire')
+        for passage in passages:
+            location = self.multiworld.get_region(f'{passage} Passage Boss', self.player).locations[0]
+            location.place_locked_item(self.create_item(f'{passage} Passage Clear'))
+            location.show_in_spoiler = False
+
+        if self.options.goal.needs_diva():
+            goal = 'Golden Diva'
+        if self.options.goal.is_treasure_hunt():
```

Comment:
These are mutually exclusive, and the only options, so `else` works just fine
```suggestion
        else:
```